push/src/main/java/org/aerogear/mobile/push/fcm/AeroGearFirebaseMessagingService.java

```diff
@@ -0,0 +1,18 @@
+package org.aerogear.mobile.push.fcm;
+
+import com.google.firebase.messaging.FirebaseMessagingService;
+import com.google.firebase.messaging.RemoteMessage;
+
+import org.aerogear.mobile.push.PushService;
+
+public class AeroGearFirebaseMessagingService extends FirebaseMessagingService {
+
+    @Override
+    public void onMessageReceived(RemoteMessage remoteMessage) {
+        super.onMessageReceived(remoteMessage);
+
+        PushService.notifyHandlers(remoteMessage.getData());
```

Comment:
Yeah so this is a good upside to the handlers being static.  We can send messages to them without needing mobile core when the message comes in and that speeds up handling.

It may be a good point of discussion for how we want to allow developers to control the various ways apps can start up.  For instance if a message comes to the app while it is killed we probably don't want to send metrics and such if all we are doing is refreshing the app...